lib/ignore/ignore.go

```diff
@@ -282,6 +292,10 @@ func (m *Matcher) ShouldIgnore(filename string) bool {
 	return false
 }
 
+func (m *Matcher) SkipIgnoredDirs() bool {
```

Comment:
I guess this is a nice optimization to have, yet I'd drop it in favor of having less code.

@calmh ?

Reply:
It's more than just optimization: E.g. the fs watcher (well notify) fails entirely if it encounters files it can't watch due to permissions, which can be worked around with ignores. I know that's far from optimal, but it is how it is. And I depend on this workaround :)